### ISSUE TITLE
Potential Issue with copying Large files to temp directory 

### ISSUE THREAD

**Issue by ThinkDigitalRepair** [sidlatau/flutter_document_picker#9]
Large Files that are copied to temp directory may fill up storage since they're being duplicated. Also, it may not be preferable to copy the files to the temp directory at all times, for example in cases where:
-  the developer wants to edit a file in place,
-  the file is large and will take up too much space, 
- the file is too large to successfully copy,
-  the file will take too long to copy
- The developer needs the original path location
- the developer needs to work with other files contained in the same directory or files that the opened file may depend on.

Long story short, can the copying to temp directory be made an optional flag that defaults to true?

**Comment by sidlatau** [comment]
Reason behind copying to temp directory is that in this case there is no need to handle Android permissions - no permission is required when reading/writing file from app temp dir. Additionally in iOS this copying is performed automatically. Also it could be problematic to get path to file otherwise - file could be in Google Drive or iCloud - what kind of paths to return to Flutter/Dart side in these cases?
For these reasons in this plugin I have chosen to copy files to temp dir in all cases. I agree that it is not optimal but it handles most common cases. I am not planning to change this functionality but pull requests are welcome!

**Comment by ThinkDigitalRepair** [comment]
For cloud cases, it can be copied, but leave the other ones optional. I'm
having the issue where my files are being overwritten because they have the
same name. (mmssms.db)even though I'm opening it from different
directories, and I have no way of differentiating them, nor accessing
anything other than the latest file. It's only a slight inconvenience now,
but I do forsee the lack of the option to be a bigger problem for others.

I'd love to submit a pull request, but it would take me too far off my
current focus to figure out how to adjust this.

Thank you for your excellent work so far. This is the next plugin I've
found to get the job done!

>

**Comment by DamonChen117** [comment]
Don’t copy the document if you can avoid it. 
From 
[https://developer.apple.com/documentation/uikit/uidocumentpickerviewcontroller](https://developer.apple.com/documentation/uikit/uidocumentpickerviewcontroller)
